pkg/kubelet/dockertools/manager.go

```diff
@@ -1579,6 +1611,16 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 			glog.Errorf("Failed to create pod infra container: %v; Skipping pod %q", err, podFullName)
 			return err
 		}
+
+		// Find the pod IP after starting the infra container in order to expose
+		// it safely via the downward API without a race
+		ip, err := dm.determineInfraContainerIP(string(podInfraContainerID), pod)
+		if err != nil {
+			glog.Errorf("Failed to determine IP of pod infra container: %v; Skipping pod %q", err, podFullName)
```

Comment:
Failing to determine pod IP does not seem like it should fail the creation of the pod.  However, in this context, if you were depending on status.PodIP and it was not set, that would be an error.  For now, simply failing is probably ok, but it does make the create sequence a tiny bit less correct.